cql3/statements/select_statement.cc

```diff
@@ -375,8 +379,8 @@ select_statement::do_execute(query_processor& qp,
     auto slice = make_partition_slice(options);
     auto max_result_size = qp.proxy().get_max_result_size(slice);
     auto command = ::make_lw_shared<query::read_command>(
-            _schema->id(),
-            _schema->version(),
+            _query_schema->id(),
+            _query_schema->version(),
```

Comment:
What's the compatibility impact? read_command is passed on the wire. What happens if we talk to an old node?

Reply:
If we have a cluster with an old node, then the read command is transformed before it is sent: a table schema is sent and the slice in the legacy-reverse form. The transformation is performed in `storege_proxy::remote::send_read_data`

Similarly, when a command is received from an old node. It is transformed into native reverse before data is read.

Reply:
IIRC reversed schema is transformed into a normal schema before pulling (based more on guessing how it works than real memory).